File: backend/src/main/java/com/revnu/backend/features/expenses/dto/ExpenseRequest.java
package com.revnu.backend.features.expenses.dto;

import java.math.BigDecimal;
import java.util.UUID;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;

public record ExpenseRequest(
        @NotNull(message = "Expense amount is required")
        @Positive(message = "Expense amount must be greater than zero")
        BigDecimal amount,
        @NotNull(message = "Category is required")
        UUID categoryId,
        @Size(max = 1000, message = "Notes is too long (max 1000 characters)")
        String notes
        ) {

}

File: mobile/app/src/main/java/com/revnu/mobile/features/expenses/model/ExpenseRequest.kt
package com.revnu.mobile.features.expenses.model

import com.google.gson.annotations.SerializedName

data class ExpenseRequest(
    @SerializedName("amount") val amount: Double,
    @SerializedName("categoryId") val categoryId: String,
    @SerializedName("notes") val notes: String?
)